lib/engine/game/g_18_zoo.rb

```diff
@@ -0,0 +1,668 @@
+          "K6": 'M',
+          "D7": 'MM',
+          "E8": 'MM',
+          "E12": 'MM',
+          "F11": 'MM',
+          "G18": 'MM',
+          "H19": 'MM',
+          "K4": 'MM',
+          "L7": 'MM',
+          "L19": 'MM',
+          "N17": 'MM',
+        },
+        map_d: {
+          "B12": 'O',
+          "B14": 'O',
+          "E19": 'O',
+          "F2": 'O',
+          "G1": 'O',
+          "G11": 'O',
+          "H10": 'O',
+          "H12": 'O',
+          "I3": 'O',
+          "K15": 'O',
+          "M13": 'O',
+          "M15": 'O',
+          "C13": 'M',
+          "D18": 'M',
+          "H16": 'M',
+          "I15": 'M',
+          "J6": 'M',
+          "D12": 'MM',
+          "E11": 'MM',
+          "F18": 'MM',
+          "G19": 'MM',
+          "J4": 'MM',
+          "K7": 'MM',
+          "K19": 'MM',
+          "M17": 'MM',
+        },
+        map_e: {
+          "D6": 'O',
+          "E17": 'O',
+          "E19": 'O',
+          "F2": 'O',
+          "G1": 'O',
+          "G11": 'O',
+          "H10": 'O',
+          "H12": 'O',
+          "I3": 'O',
+          "K15": 'O',
+          "M13": 'O',
+          "M15": 'O',
+          "C13": 'M',
+          "D18": 'M',
+          "H16": 'M',
+          "I15": 'M',
+          "J6": 'M',
+          "C7": 'MM',
+          "D12": 'MM',
+          "E11": 'MM',
+          "F18": 'MM',
+          "G19": 'MM',
+          "J4": 'MM',
+          "K7": 'MM',
+          "K19": 'MM',
+          "M17": 'MM',
+        },
+        map_f: {
+          "C12": 'O',
+          "C14": 'O',
+          "D9": 'O',
+          "E6": 'O',
+          "F19": 'O',
+          "G2": 'O',
+          "H1": 'O',
+          "H11": 'O',
+          "I10": 'O',
+          "I12": 'O',
+          "J3": 'O',
+          "L15": 'O',
+          "L17": 'O',
+          "D13": 'M',
+          "E18": 'M',
+          "F9": 'M',
+          "I16": 'M',
+          "K6": 'M',
+          "D7": 'MM',
+          "E8": 'MM',
+          "E12": 'MM',
+          "F11": 'MM',
+          "G18": 'MM',
+          "H19": 'MM',
+          "K4": 'MM',
+          "L7": 'MM',
+        },
+      }.freeze
+
+      HOME_TOKEN_TIMING = :float
+
+      MUST_BUY_TRAIN = :always
+
+      # A yellow/upgrade and a yellow
+      TILE_LAYS = [{ lay: true, upgrade: true }, { lay: true, upgrade: :not_if_upgraded }].freeze
+
+      MARKET_TEXT = Base::MARKET_TEXT.merge(par_2: 'Can only enter during phase green',
+                                            par_3: 'Can only enter during phase brown').freeze
+
+      MARKET_SHARE_LIMIT = 80 # percent
+
+      attr_reader :available_companies, :future_companies
+
+      def setup
+        @operating_rounds = 2 # 2 ORs on first and second round
+
+        @available_companies = []
+        @future_companies = []
+
+        draw_size = players.size == 5 ? 6 : 4
+        @companies_for_isr = @companies.first(draw_size)
+        @companies_for_monday = @companies.drop(draw_size + 4).first(4)
```

Comment:
are you trying to slide here?

@companies[draw_size..draw_size + 4)

Reply:
done in 558751e5